legend-shared-pac4j/src/main/java/org/finos/legend/server/pac4j/hazelcaststore/HazelcastSessionStore.java

```diff
@@ -0,0 +1,157 @@
+            if (mapConfigs.size() == 1 && optionalMapConfig.get().getTimeToLiveSeconds() != 0)
+            {
+                MapConfig hazelcastMapConfig = optionalMapConfig.get();
+                this.hazelcastMap = hazelcastInstance.getMap(hazelcastMapConfig.getName());
+                this.maxSessionLength = hazelcastMapConfig.getTimeToLiveSeconds();
+            }
+            else
+            {
+                throw new IllegalStateException(
+                        "The Hazelcast config needs to include exactly one Map Configuration with a TTL seconds value");
+            }
+        }
+        catch (FileNotFoundException e)
+        {
+            throw new UncheckedIOException(
+                    "Failed to find Hazelcast config file in specified path: " + hazelcastConfigFilePath, e);
+        }
+    }
+
+    private SessionToken getOrCreateSsoKey(WebContext context)
+    {
+        SessionToken token = SessionToken.fromContext(context);
+        if (token == null)
+        {
+            token = createSsoKey(context);
+        }
+        return token;
+    }
+
+    private SessionToken createSsoKey(WebContext context)
+    {
+        SessionToken token = SessionToken.generate();
+        token.saveInContext(context, maxSessionLength);
+        Map<String, Object> hazelcastSessionData = new HashMap<>();
+        hazelcastMap.put(token.getSessionId(), hazelcastSessionData);
+        return token;
+    }
+
+    @Override
+    public String getOrCreateSessionId(WebContext context)
+    {
+        getOrCreateSsoKey(context);
+        return super.getOrCreateSessionId(context);
+    }
+
+    @Override
+    public Object get(WebContext context, String key)
+    {
+        Object res = super.get(context, key);
+        if (res == null)
+        {
+            SessionToken token = getOrCreateSsoKey(context);
+            Map<String, Object> hazelcastSessionData = hazelcastMap.get(token.getSessionId());
+            if (hazelcastSessionData != null)
+            {
+                Object data = hazelcastSessionData.get(key);
+                if (data != null)
+                {
+                    res = data;
+                    //Once we have it, store it in the regular session store for later access
+                    super.set(context, key, res);
+                }
+            }
+        }
+        else if (SessionToken.fromContext(context) == null)
+        {
+            //if res is not null ,this means we still have an active Session but an expired SSO cookie we need to recreate one and add it to the context request/response.
+            createSsoKey(context);
+            if (res instanceof LinkedHashMap)
+            {
+                ProfileHelper.flatIntoAProfileList((LinkedHashMap<String, CommonProfile>)res);
+                set(context, Pac4jConstants.USER_PROFILES, ProfileHelper.flatIntoAProfileList((LinkedHashMap<String, CommonProfile>)res));
+            }
```

Comment:
There are two calls to ProfileHelper.flatIntoAProfileList, and the first one seems to be useless. Also, why do we care if res is specifically a LinkedHashMap? Why won't any Map do?
```suggestion
            if (res instanceof LinkedHashMap)
            {
                set(context, Pac4jConstants.USER_PROFILES, ProfileHelper.flatIntoAProfileList((LinkedHashMap<String, CommonProfile>)res));
            }
```

Reply:
Removed the unnecessary call. Keeping the rest the same for now